integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/BeforeAfterMessagingProxyExecutor.java

```diff
@@ -0,0 +1,30 @@
+package com.sequenceiq.it.cloudbreak.newway;
+
+import com.sequenceiq.it.cloudbreak.exception.ProxyMethodInvocationException;
+
+import javax.ws.rs.WebApplicationException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.Callable;
+
+import static com.sequenceiq.it.cloudbreak.newway.log.Log.log;
+
+public final class BeforeAfterMessagingProxyExecutor extends GenericProxyExecutor {
+
+    @Override
+    public <R> R exec(Callable<R> method) {
+        try {
+            return method.call();
+        } catch (InvocationTargetException ite) {
+            if (ite.getCause() instanceof WebApplicationException) {
+                String content = WebApplicationException.class.cast(ite.getCause()).getResponse().readEntity(String.class);
+                log(content);
+                throw WebApplicationException.class.cast(ite.getCause());
```

Comment:
Can i ask why do you prefer that long cast instead of `(WebApplicationException) ite.getCause()`?

Reply:
long? just 8 character longer than your solution (since you need an extra pair of brackets). does it make anything less readable or makes any performance issue?